plugins/woocommerce/src/Admin/Features/PaymentGatewaySuggestions/EvaluateSuggestion.php

```diff
@@ -72,9 +85,25 @@ public static function evaluate_specs( $specs, $logger_args = array() ) {
 			}
 		}
 
-		return array(
+		$result = array(
 			'suggestions' => $suggestions,
 			'errors'      => $errors,
 		);
+
+		// Memoize results, with a fail safe to prevent unbounded memory growth.
+		// This limit is unlikely to be reached under normal circumstances.
+		if ( count( self::$memo ) > 50 ) {
+			self::$memo = array();
```

Comment:
Nit: should we reuse the `reset_memo` function here?

Reply:
Sure, updated in df27eaac12ee13e2ae014297be001f0832e3f75e